packages/editor/src/components/block-toolbar/index.js

```diff
@@ -34,9 +34,9 @@ function BlockToolbar( { blockClientIds, isValid, mode } ) {
 					<BlockSwitcher clientIds={ blockClientIds } />
 					<BlockControls.Slot />
 					<BlockFormatControls.Slot />
+					<BlockSettingsMenu clientIds={ blockClientIds } />
 				</Fragment>
 			) }
-			<BlockSettingsMenu clientIds={ blockClientIds } />
```

Comment:
I'm not sure why it wasn't considered before to hide this if the block is invalid, so I'd want some perspective on the potential implications of this.

Reply:
I've learned why: 

![peek 2018-11-22 11-19](https://user-images.githubusercontent.com/583546/48896691-861e7880-ee48-11e8-89e1-f9b61322c58f.gif)